### PR TITLE
Intro setup

### PR DESCRIPTION
**Pull request by cm97878**
Added intro "cutscenes" and moved cutscene handling to its own store. Also added a toggle for whether saves load on startup or not, as well as a game stage tracker for general progression-dependent changes.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -42,29 +45,35 @@
...
-                :disabled="player.getSoul.lt(player.getMaxSoul)">{{ !player.tails.obtained ? '???' : 'Gain Tail' }}</button>
+                v-show="(player.tails > 1 && player.tails < 9) || (player.tails === 1 && player.getSoul.eq(player.getMaxSoul))" 
+                :disabled="player.getSoul.lt(player.getMaxSoul)">{{ player.furthestStage <= GameStage.PRE_TAILS ? '???' : 'Gain Tail' }}</button>
             </div>
+
             <div class="options-box">
                 <button @click="saves.save()">Save</button>
                 <button @click="saves.load()">Load</button>
-                <button @click="player.addSoul(1000000000000000);">add max soul</button>
+                <button @click="player.addSoul(10000000000000000);">add max soul</button>
```
Comment (Malthusis) [inline src/App.vue:69]: Not sure why you added this back in, the number is too large to be a `number`: 
image

File: src/components/CutsceneModal.vue [hunk 2]
```diff
@@ -2,21 +2,32 @@
...
-    import type { EventChoice }  from '@/types/areaEvent'
-    const props = defineProps<{
-      text: string,
-      choices: EventChoice[],
-    }>()
-    const emit = defineEmits(['choice'])
+    import { useEventStore } from '@/stores/eventStore';
+    import { storeToRefs } from 'pinia';
+import { computed } from 'vue';
+
```
Comment (Malthusis) [inline src/components/CutsceneModal.vue:17]: fix indentation here.

File: src/stores/eventStore.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        if(!activeScene.value?.chain) {
+            activeScene.value = undefined;
+        }
+    }
+
+    // function cutsceneResponse(choice: number): void {
+    //     //for now, cause non eligble choice to be 2.
+    //     choice === 1 ? combatStore.pushToCombatLog("You chose number 1!") : combatStore.pushToCombatLog("You chose number 2!")
+    //     cutsceneActive.value = false;
+    // }
```
Comment (Malthusis) [inline src/stores/eventStore.ts:170]: This code still needed?

File: src/stores/eventStore.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    const combatStore = useCombatStore();
+    const mapStore = useMapStore();
+    const player = usePlayer();
+
+    const activeScene = ref<Cutscene>();
+    const sceneDesc = ref("");
+    const soulDenyCounter = ref(0);
+
+
+    const cutscenes = new Map<string, Cutscene>([
```
Comment (Malthusis) [inline src/stores/eventStore.ts:20]: I'd put a TODO here before the cutscenes struct on moving the cutscenes out to a JSON file for easier editing at some future point.

**Comment by Malthusis** [comment]
Looks solid!

File: src/App.vue (hunk 1 above)
Comment (cm97878) [reply src/App.vue:69]: oh whoops, didnt add it in so much as just didnt merge that change when i pulled from master, ill get that fixed

File: src/stores/eventStore.ts (hunk 3 above)
Comment (cm97878) [reply src/stores/eventStore.ts:170]: nope

**Pull request merged**